pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -743,15 +743,20 @@ type TriggerParameterSource struct {
 	// To access an array value use the index as the key. The dot and wildcard characters can be escaped with '\\'.
 	// See https://github.com/tidwall/gjson#path-syntax for more information on how to use this.
 	DataKey string `json:"dataKey,omitempty" protobuf:"bytes,4,opt,name=dataKey"`
+	// UseRawDataValue indicates if the value in an event at data key should be used without converting to string.
+	// When true, a number, boolean, json or string parameter may be extracted. When the field is unspecified, or explicitly
+	// false, the behavior is to turn the extracted field into a string or json. (e.g. when set to true, the parameter
+	// 123 will resolve to the numerical type, but when false, or not provided, the string "123" will be resolved)
+	UseRawDataValue bool `json:"useRawDataValue,omitempty" protobuf:"bytes,5,opt,name=useRawDataValue"`
```

Comment:
1. Please add it to the end of the struct - do not change the order of original properties.

2. If it not specified, consider it as the `string` (reverting https://github.com/argoproj/argo-events/pull/2273), in this case, if expecting a JSON, just set it to `true`.

Reply:
👍 made the change. let me know if the wording of this comment can be improved.